cvat/apps/engine/serializers.py

```diff
@@ -404,10 +405,21 @@ def update_label(
         except models.InvalidLabel as exc:
             raise exceptions.ValidationError(str(exc)) from exc
 
+        mapped_data = {}
+        for label_data in initial_data.get('labels'):
+            for attr_data in label_data.get('attributes', []):
+                mapped_data[attr_data.get('name', None)] = attr_data.get('id', None)
+
         for attr in attributes:
-            (db_attr, created) = models.AttributeSpec.objects.get_or_create(
-                label=db_label, name=attr['name'], defaults=attr
-            )
+            attr_id = mapped_data.get(attr['name'], None)
+            if attr_id is not None:
+                (db_attr, created) = models.AttributeSpec.objects.get_or_create(
+                    label=db_label, id=attr_id, defaults=attr
+                )
```

Comment:
Yes, instead of database, modifying the serializer is a much better option.

Reply:
Here I don't think we need to use try-catch block. This is because in-case of creating new attributes, there is no id received and so it can be none.
I think this would make more sense.
![image](https://github.com/opencv/cvat/assets/76646497/7d494977-89f6-4cb7-941c-ff7de812b9a0)

Reply:
@ritikraj26, I'm not telling you to remove the code that creates or defines `db_attr` based on `attr['name']`.
I just meant that if the attribute `id` was provided in the request body, then we should not use the `get_or_create` method, but should check that the attribute with that id exists. If there is no attribute with a specified `id`, then we need to throw an exception and return the 400(or better 404) code to the client.

So the correct code block will be
```python
for attr in attributes:
    attr_id = attr.get('id')
    if attr_id is not None:
        try:
            db_attr = models.AttributeSpec.objects.get(id=attr_id)
        except models.AttributeSpec.DoesNotExist as ex:
            raise exceptions.NotFound(f'The attribute with id #{attr_id} does not exist') from ex
        created = False
    else:
        (db_attr, created) = models.AttributeSpec.objects.get_or_create(
            label=db_label, name=attr['name'], defaults=attr
        )
```

Reply:
Thanks, this makes it very much clear.